scilpy/tractanalysis/scoring.py

```diff
@@ -0,0 +1,389 @@
+        tmp_sft, mask_2, 'either_end', False)
+    remaining_ids = np.setdiff1d(range(len(sft.streamlines)),
+                                 mask_1_ids[list(mask_2_ids)])
+    remaining_sft = sft[list(remaining_ids)]
+
+    return extracted_sft, remaining_sft
+
+
+def get_binary_maps(streamlines, dimensions, sft, invalid):
+    """
+    Extract a mask from a bundle
+
+    Parameters
+    ----------
+    streamlines: list
+        List of streamlines.
+    dimensions: tuple of ints
+        Dimensions of the mask.
+    sft : StatefulTractogram
+        Reference tractogram.
+    invalid: bool
+        If true, remove invalid streamlines from tractogram.
+
+    Returns
+    -------
+    bundles_voxels: numpy.ndarray
+        Mask representing the bundle volume.
+    endpoints_voxels: numpy.ndarray
+        Mask representing the bundle's endpoints.
+    """
+
+    if not len(streamlines):
+        return np.zeros(dimensions), np.zeros(dimensions)
+    elif len(streamlines) == 1:
+        streamlines = [streamlines]
+    tmp_sft = StatefulTractogram.from_sft(streamlines, sft)
+    tmp_sft.to_vox()
+    tmp_sft.to_corner()
+    if invalid:
+        tmp_sft.remove_invalid_streamlines()
+
+    if len(tmp_sft) == 1:
+        return np.zeros(dimensions), np.zeros(dimensions)
+
+    bundles_voxels = compute_tract_counts_map(tmp_sft.streamlines,
+                                              dimensions).astype(np.int16)
+
+    endpoints_voxels = get_endpoints_density_map(tmp_sft.streamlines,
+                                                 dimensions).astype(np.int16)
+
+    bundles_voxels[bundles_voxels > 0] = 1
+    endpoints_voxels[endpoints_voxels > 0] = 1
+
+    return bundles_voxels, endpoints_voxels
+
+
+def compute_gt_masks(gt_bundles, parser, args):
+    """
+    Compute ground-truth masks. If the ground-truth is
+    already a mask, load it. If the ground-truth is a
+    bundle, compute the mask.
+
+    Parameters
+    ----------
+    gt_bundles: list
+        List of either StatefulTractograms or niftis.
+    parser: ArgumentParser
+        Argument parser which handles the script's arguments.
+    args: Namespace
+        List of arguments passed to the script.
+
+    Returns
+    -------
+    mask_1: numpy.ndarray
+        "Head" of the mask.
+    mask_2: numpy.ndarray
+        "Tail" of the mask.
+    """
+
+    gt_bundle_masks = []
+    gt_bundle_inv_masks = []
+
+    for gt_bundle in args.gt_bundles:
+        # Support ground truth as streamlines or masks
+        # Will be converted to binary masks immediately
+        _, ext = split_name_with_nii(gt_bundle)
+        if ext in ['.gz', '.nii.gz']:
+            gt_img = nib.load(gt_bundle)
+            gt_mask = get_data_as_mask(gt_img)
+            affine = gt_img.affine
+            dimensions = gt_mask.shape
+        else:
+            gt_sft = load_tractogram_with_reference(
+                parser, args, gt_bundle, bbox_check=False)
+            if args.remove_invalid:
```

Comment:
Move this if two lines down, this function goes to vox/corner and back to its initial state.
So if you call remove_invalid_streamlines already in vox/corner you save a transformation

Reply:
Removed as per previous comment